quinn-udp/src/lib.rs

```diff
@@ -238,3 +238,23 @@ impl EcnCodepoint {
         })
     }
 }
+
+#[cfg(target_os = "linux")]
+#[derive(Clone, Debug, Copy)]
+pub struct IcmpError {
+    pub dst: SocketAddr,
+    pub kind: IcmpErrorKind,
+}
+
+#[cfg(not(target_os = "linux"))]
+pub struct IcmpError;
```

Comment:
Having the struct not exist on non-linux shouldn't be necessary, and requires consumers of the library to do platform-specific conditional compilation. Unless I'm misinterpreting, there should just be a single non-conditional `IcmpError` struct and `IcmpErrorKind` struct, and conditional compilation can be used to make functions which return them always return `Err` or `None` or something on non-linux.